security/pkg/nodeagent/sds/sdsservice_test.go

```diff
@@ -70,6 +73,60 @@ func TestStreamSecretsForWorkloadSds(t *testing.T) {
 	testHelper(t, arg, sdsRequestStream, false)
 }
 
+func TestStreamSecretsForLocalJWTGatewaySds(t *testing.T) {
+	cf, err := plugin.NewMockCredFetcher(
+		security.GCE, "abc.svc.id.goog", "/var/run/secrets/tokens/istio-token")
+	if err != nil {
+		t.Errorf("unexpected Error: %v", err)
+	}
+	arg := ca2.Options{
+		EnableGatewaySDS:  true,
+		EnableWorkloadSDS: false,
+		RecycleInterval:   30 * time.Second,
+		GatewayUDSPath:    fmt.Sprintf("/tmp/gateway_gotest%q.sock", string(uuid.NewUUID())),
+		WorkloadUDSPath:   "",
+		UseLocalJWT:       true,
+		CredFetcher:       cf,
+	}
+	testHelper(t, arg, sdsRequestStream, false)
+}
+
+func TestStreamSecretsForLocalJWTWorkloadSds(t *testing.T) {
```

Comment:
Please add a brief comment explaining the test here and other tests below. The purpose of adding these tests is to verify that SDS agent is using the token returned by credential fetcher. Please update the test names to make it explicit.